raiden/api/validations/channel_validator.py

```diff
@@ -139,17 +139,28 @@ def validate_and_get_channels_to_settle(token_address: TokenAddress,
         if token_address not in valid_tokens:
             raise UnknownTokenAddress("Token address is not known in channel settle.")
 
-        channels = views.get_channelstate_settling(
-            chain_state=chain_state,
-            payment_network_id=registry_address,
-            token_address=token_address,
-            creator_address=creator_address
-        )
-
         channels_to_settle: List[NettingChannelState] = []
 
-        for channel in channels:
-            if channel.partner_state.address == partner_address:
-                channels_to_settle.append(channel)
+        channels = filter(lambda channel: channel.partner_state.address == partner_address,
+                          views.get_channelstate_settling(
+                              chain_state=chain_state,
+                              payment_network_id=registry_address,
+                              token_address=token_address,
+                              creator_address=creator_address
+                          ))
+
+        channels_to_settle.extend(channels)
+
+        # we look for the same but with the creator as the partner since we can
+        # be in the other side and the creator of the channel isn't our address
```

Comment:
this comment isn't so clear. could you please write it in spanish so i could understand it better? (and maybe come up with a suggestion)

Reply:
Let me change it to something better